test/TestSpan.cpp

```diff
@@ -274,8 +304,41 @@ TEST(span, serialize_json)
     span.serialize_json(writer);
 
     char str[2048] = {0};
-    int str_len = snprintf(str, sizeof(str), json_template, span.trace_id_high(), span.trace_id(), span.id(), span.parent_id(),
-                           span.message().annotations[0].timestamp, span.message().timestamp);
+    int str_len = snprintf(str, sizeof(str), json_template, span.trace_id_high(), span.trace_id(), span.id(),  span.parent_id(), span.message().annotations[0].timestamp, span.message().timestamp);
+
+    ASSERT_EQ(std::string(buffer.GetString(), buffer.GetSize()), std::string(str, str_len));
+}
+
+TEST(span, serialize_json_v2)
```

Comment:
probably want to add some more tests since there's a lot of conversion logic. Perhaps port some from here for the major cases https://github.com/openzipkin/zipkin/blob/master/zipkin/src/test/java/zipkin/internal/Span2ConverterTest.java